libs/utils/tests/postgres_backend_async.rs

```diff
@@ -0,0 +1,140 @@
+/// Test postgres_backend_async with tokio_postgres
+use once_cell::sync::Lazy;
+use pq_proto::{BeMessage, RowDescriptor};
+use std::io::Cursor;
+use std::{future, sync::Arc};
+use tokio::net::{TcpListener, TcpStream};
+use tokio_postgres::config::SslMode;
+use tokio_postgres::tls::MakeTlsConnect;
+use tokio_postgres::{Config, NoTls, SimpleQueryMessage};
+use tokio_postgres_rustls::MakeRustlsConnect;
+use utils::postgres_backend::{AuthType, Handler, PostgresBackend, QueryError};
+
+// generate client, server test streams
+async fn make_tcp_pair() -> (TcpStream, TcpStream) {
+    let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let addr = listener.local_addr().unwrap();
+    let client_stream = TcpStream::connect(addr).await.unwrap();
+    let (server_stream, _) = listener.accept().await.unwrap();
+    (client_stream, server_stream)
+}
```

Comment:
https://docs.rs/tokio/latest/tokio/io/fn.duplex.html

Reply:
Cool, have been searching for that and didn't find.

Reply:
... except that currently pgbackend is not parametrized with the stream. There is PR in progress by Arthur on that, we can change this there. And this is old code FWIW.

Reply:
Is it already changed there? If not lets add an issue with tech_debt tag. This also looks like a separate change that can be done in completely separate PR.

Reply:
I'd say that's the case when creating/managing the issue is much longer than the actual fix (once parametrization lands) and makes managing/triaging more important issues hard, but here you go.
https://github.com/neondatabase/neon/issues/3731